include/libopencm3/sam/4l/pm.h

```diff
@@ -115,71 +105,74 @@
 
 /* --- Register contents --------------------------------------------------- */
 #define PM_MCCTRL_MCSEL_SHIFT			0
-
-#define PM_CPUSEL_CPUDIV_SHIFT			7
-#define PM_CPUSEL_CPUSEL_SHIFT			0
-
-#define PM_PBSEL_PBDIV_SHIFT			7
-#define PM_PBSEL_PBSEL_SHIFT			0
-
-#define PM_CPUMASK_OSC_SHIFT			0
-
-#define PM_HSBMASK_PDCA_SHIFT			0
-#define PM_HSBMASK_FLASHCALW_SHIFT			1
-#define PM_HSBMASK_FLASHCALW_PICO_SHIFT			2
-#define PM_HSBMASK_USBC_SHIFT			3
-#define PM_HSBMASK_CRCCU_SHIFT			4
-#define PM_HSBMASK_APBA_SHIFT			5
-#define PM_HSBMASK_APBB_SHIFT			6
-#define PM_HSBMASK_APBC_SHIFT			7
-#define PM_HSBMASK_APBD_SHIFT			8
-#define PM_HSBMASK_AESA_SHIFT			9
-
-#define PM_PBAMASK_IISC_SHIFT			0
-#define PM_PBAMASK_SPI_SHIFT			1
-#define PM_PBAMASK_TC0_SHIFT			2
-#define PM_PBAMASK_TC1_SHIFT			3
-#define PM_PBAMASK_TWIM0_SHIFT			4
-#define PM_PBAMASK_TWIS0_SHIFT			5
-#define PM_PBAMASK_TWIM1_SHIFT			6
-#define PM_PBAMASK_TWIS1_SHIFT			7
-#define PM_PBAMASK_USART0_SHIFT			8
-#define PM_PBAMASK_USART1_SHIFT			9
-#define PM_PBAMASK_USART2_SHIFT			10
-#define PM_PBAMASK_USART3_SHIFT			11
-#define PM_PBAMASK_ADCIFE_SHIFT			12
-#define PM_PBAMASK_DACC_SHIFT			13
-#define PM_PBAMASK_ACIFC_SHIFT			14
-#define PM_PBAMASK_GLOC_SHIFT			15
-#define PM_PBAMASK_ABDACB_SHIFT			16
-#define PM_PBAMASK_TRNG_SHIFT			17
-#define PM_PBAMASK_PARC_SHIFT			18
-#define PM_PBAMASK_CATB_SHIFT			19
+#define PM_MCCTRL_MCSEL_MASK			3
+
+// Values common for CPUSEL and PBxSEL
+#define PM_CKSEL_DIV			(1 << 7)
+#define PM_CKSEL_MASK			(3)
+
+#define PM_CPUMASK_OSC			(1 << 0)
+
+#define PM_HSBMASK_PDCA			(1 << 0)
+#define PM_HSBMASK_FLASHCALW			(1 << 1)
+#define PM_HSBMASK_FLASHCALW_PICO			(1 << 2)
+#define PM_HSBMASK_USBC			(1 << 3)
+#define PM_HSBMASK_CRCCU			(1 << 4)
+#define PM_HSBMASK_APBA			(1 << 5)
+#define PM_HSBMASK_APBB			(1 << 6)
+#define PM_HSBMASK_APBC			(1 << 7)
+#define PM_HSBMASK_APBD			(1 << 8)
+#define PM_HSBMASK_AESA			(1 << 9)
+
+#define PM_PBAMASK_IISC			(1 << 0)
+#define PM_PBAMASK_SPI			(1 << 1)
+#define PM_PBAMASK_TC0			(1 << 2)
+#define PM_PBAMASK_TC1			(1 << 3)
+#define PM_PBAMASK_TWIM0			(1 << 4)
+#define PM_PBAMASK_TWIS0			(1 << 5)
+#define PM_PBAMASK_TWIM1			(1 << 6)
+#define PM_PBAMASK_TWIS1			(1 << 7)
+#define PM_PBAMASK_USART0			(1 << 8)
+#define PM_PBAMASK_USART1			(1 << 9)
+#define PM_PBAMASK_USART2			(1 << 10)
+#define PM_PBAMASK_USART3			(1 << 11)
+#define PM_PBAMASK_ADCIFE			(1 << 12)
+#define PM_PBAMASK_DACC			(1 << 13)
+#define PM_PBAMASK_ACIFC			(1 << 14)
+#define PM_PBAMASK_GLOC			(1 << 15)
+#define PM_PBAMASK_ABDACB			(1 << 16)
+#define PM_PBAMASK_TRNG			(1 << 17)
+#define PM_PBAMASK_PARC			(1 << 18)
+#define PM_PBAMASK_CATB			(1 << 19)
```

Comment:
heh, ok.  And I'd just left comments on exactly that from the previous commit.  I'm going to stop here and look again later.